### ISSUE TITLE
Aggregation

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-inc/quickwit#311]
We want to have some flavor of aggregation.

**Comment by mosuka** [comment]
This is a proposal for an aggregation REST API definition.
Aggregation works as a subset of the search API. To perform aggregation, add the `aggregation` parameter to the `search` API.

The REST API specification is as follows:

## Endpoint
```
GET api/v1/indexes/<index name>/search
```

## Request body
```
{
    "aggretations": {
        "<aggregation name>": {
            "<aggregation type>": {
                "<aggregation parameter name>": <aggregation parameter value>,
                "<aggregation parameter name>": <aggregation parameter value>,
                ...
            }
        }
    }
}
```

| Name | Description |
| --- | --- |
| aggregation name | Aggregation name. Give it a name that will be unique within the `aggregations`. |
| aggregation type | Specifies the type of aggregation to be performed. |
| aggregation parameter name | The name of the parameter given to the aggregation to be performed. |
| aggregation parameter value | The value of the parameter given to the aggregation to be performed. |

## Respons body
```
{
    "aggregations": {
        "<aggregation name>": {
            "<aggregation result name>": <aggregation result value>, 
            "<aggregation result name>": <aggregation result value>,
            ...
        }
    }
}
```

| Name | Description |
| --- | --- |
| aggregation name | Aggregation name specified in the request. |
| aggregation result name | Name of the result of the aggregation performed. |
| aggregation result value | Value of the result of the aggregation performed. |

**Comment by fulmicoton** [comment]
Closes in favor for #832
